main.go

```diff
@@ -195,9 +195,39 @@ func main() {
 	zoneTypeFilter := provider.NewZoneTypeFilter(cfg.AWSZoneType)
 	zoneTagFilter := provider.NewZoneTagFilter(cfg.AWSZoneTagFilter)
 
-	var awsSession *session.Session
+	awsSessions := make(map[string]*session.Session, len(cfg.AWSProfiles))
+	var awsDefaultSession *session.Session
 	if cfg.Provider == "aws" || cfg.Provider == "aws-sd" || cfg.Registry == "dynamodb" {
-		awsSession, err = aws.NewSession(
+		if len(cfg.AWSProfiles) == 0 || (len(cfg.AWSProfiles) == 1 && cfg.AWSProfiles[0] == "") {
```

Comment:
Here is a longer code change without test.
Maybe better to also have in our provider/aws package a function that creates the session instead of putting it in to main(). main() is not good testable.

Reply:
Hi @szuecs, I extracted two functions, one to create sessions per profile and one to create a default session. I hope this is wat you've intended. Anyway I stumbled over two aspects:
1. I think the code of creating sessions over creating AWS clients to create an AWS provider can be further streamlined to remove complexity from main. I think this can all be integrated in `aws.NewAWSProvider()`. I'm eager to work on that but I guess its better to do that in a dedicated PR. WDYT?
2. I didn't add tests for the newly created functions. IMHO they're hard to unit test due to their strong dependencies to AWS SDK packages and behavior. However, I was wondering if you ever considered making use of LocalStack for testing the AWS provider? Same here, I would be eager to take care about that.

Reply:
I could not find any suitable mocking option for AWS sessions in AWS SDK v1, what is still in use here. However, it turned out that in most cases session creation facilities in AWS SDK don't talk to any remote endpoint. So I was able to test the profile loading behavior from a credential file as well as the default behavior from before this PR. What is unfortunately not possible to test in this way is the role assumption options. They require connectivity to AWS' STS service.

Reply:
ok let's go with that!